src/search_stream.rs

```diff
@@ -203,6 +213,9 @@ impl<'a, R: io::Read, W: Terminal + Send> Searcher<'a, R, W> {
                 break;
             }
             while self.inp.pos < self.inp.lastnl {
+                if self.opts.files_with_matches && self.match_count > 0 {
+                    break;
```

Comment:
It can't be *right* after it because the first match needs to be handled. But what's odd is that if I put it at the end of the while loop, I get this:

```rust
loop {
            println!("{:?} {}", self.path, self.match_count);
...
while ... {
...
        if self.opts.files_with_matches && matched {
            break;
        }
    }
}
```
```
"src/args.rs" 1
"src/args.rs" 2
"src/args.rs" 3
"src/args.rs" 3
```

I think I have to break out of both loops maybe.